packages/menu/src/MenuItem.ts

```diff
@@ -276,6 +324,108 @@ export class MenuItem extends LikeAnchor(Focusable) {
         }
     }
 
+    public closeOverlay?: (leave?: boolean) => Promise<void>;
+
+    protected handleSubmenuClick(): void {
+        this.openOverlay({ immediate: true });
+    }
+
+    protected handlePointerenter(): void {
+        if (this.leaveTimeout) {
+            clearTimeout(this.leaveTimeout);
+            delete this.leaveTimeout;
+        } else {
+            this.openOverlay();
+        }
+    }
+
+    protected leaveTimeout?: ReturnType<typeof setTimeout>;
+
+    protected handlePointerleave(): void {
+        if (this.hasSubmenu && this.open) {
+            this.active = true;
+            this.leaveTimeout = setTimeout(() => {
+                delete this.leaveTimeout;
+                if (this.closeOverlay) this.closeOverlay(true);
+            }, POINTERLEAVE_TIMEOUT);
+        }
+    }
+
+    protected handleSubmenuChange = (): void => {
+        this.click();
+    };
+
+    protected handleSubmenuPointerenter = (): void => {
+        if (this.leaveTimeout) {
+            clearTimeout(this.leaveTimeout);
+            delete this.leaveTimeout;
+        }
+    };
+
+    public async openOverlay({
+        immediate,
+    }: { immediate?: boolean } = {}): Promise<void> {
+        if (this.hasSubmenu && !this.open) {
+            this.open = true;
+            this.submenu = (
+                this.shadowRoot.querySelector(
+                    'slot[name="submenu"]'
+                ) as HTMLSlotElement
+            ).assignedElements()[0] as Menu;
```

Comment:
Maybe an opportunity for https://lit.dev/docs/api/decorators/#queryAssignedElements?

Reply:
We aren't technically at that version yet, and a quick attempt ran into some TS annoyances, so punting on this.